kong/dao/dao.lua

```diff
@@ -35,6 +38,16 @@ local function check_not_empty(tbl, arg_n)
   end
 end
 
+local function check_utf8(tbl, arg_n)
+  if tbl then
```

Comment:
side note: this check is relevant because you always call `check_utf8` after `check_arg` (as it should indeed be the case), hence `tbl` is guaranteed not to be `nil` and to be a table.